script.js

```diff
@@ -0,0 +1,79 @@
+const h2El = document.querySelector('h2');
+const numBtnEl = document.querySelectorAll('.number-button');
+const opBtnEl = document.querySelectorAll('.operation-button');
+const eqBtnEl = document.querySelector('.equals-button');
+const resetBtnEl = document.querySelector('.reset-button');
+
+let total = 0
+
+function displayNum(e) {
+    if (total === 0) {
+        const text = e.target.id
+        display = h2El.innerText + text
+        h2El.innerText = display
+    } else if (h2El.classList.contains('clear')){
+        h2El.innerText = ''
+        const text = e.target.id
+        display = h2El.innerText + text
+        h2El.innerText = display
+        h2El.classList.remove('clear')
+    } else {
+        const text = e.target.id
+        display = h2El.innerText + text
+        h2El.innerText = display
+    }
+}
+
+numBtnEl.forEach(btn => {
```

Comment:
opt for elegation over an On operation (array search) since it essentially creates a new function on every el in your array in memory